cmd/kaspawallet/daemon/server/sync.go

```diff
@@ -103,16 +112,24 @@ func (s *server) maxUsedIndex() uint32 {
 
 // collectRecentAddresses collects addresses from used addresses until
 // the address with the index of the last used address + 1000.
-// collectRecentAddresses scans addresses in batches of numIndexesToQuery,
+// collectRecentAddresses scans addresses in batches of 1000,
 // and releases the lock between scans.
 func (s *server) collectRecentAddresses() error {
-	maxUsedIndex := s.maxUsedIndex()
-	for i := uint32(0); i < maxUsedIndex+1000; i += numIndexesToQuery {
-		err := s.collectAddressesWithLock(i, i+numIndexesToQuery)
+	index := uint32(0)
+	maxUsedIndex := uint32(0)
+	for ; index < maxUsedIndex+1000; index += 1000 {
+		err := s.collectAddressesWithLock(index, index+1000)
 		if err != nil {
 			return err
 		}
+		maxUsedIndex = s.maxUsedIndex()
+	}
+
+	s.lock.Lock()
+	if index > s.nextSyncStartIndex {
+		s.nextSyncStartIndex = index
```

Comment:
Maybe the name doesn't reflect it, but `s.nextSyncStartIndex` is used only for `collectFarAddresses`, so I don't understand why would you want to reset it to the last checked recent address

Reply:
it is the main fix of slow sync. as it used in isSynced() 
otherwise we need to wait until collectFarAddresses() will reach it